src/http/headers.rs

```diff
@@ -0,0 +1,137 @@
+use crate::ffi::{
+    ngx_http_headers_in_t, ngx_http_headers_out_t, ngx_list_part_t, ngx_list_t, ngx_pool_t, ngx_table_elt_t,
+    ngx_uint_t, u_char,
+};
+use crate::ngx_string;
+use std::ffi::CString;
+use std::fmt;
+use std::ptr;
+
+pub struct Header(*mut ngx_table_elt_t, *mut ngx_pool_t);
+
+impl Header {
+    pub fn set_value(&mut self, value: &str) {
+        // we can use pool to allocate and then copy
+        unsafe { self.0.as_mut() }.map(|table| {
+            table.value.len = value.len() as _;
+            table.value.data = crate::ffi::str_to_uchar(self.1, value);
+        });
+
+        // Alternative way is using CString and transfer ownership.
```

Comment:
Any speculation which one might be more efficient?

I'm wagering CString will put memory management under Rust semantics and reference counting. Where pool will allocate from nginx, could this mean it gets held longer in the pool?